hbase-hbtop/src/main/java/org/apache/hadoop/hbase/hbtop/Record.java

```diff
@@ -0,0 +1,180 @@
+ */
+package org.apache.hadoop.hbase.hbtop;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.EnumMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Stream;
+import javax.validation.constraints.NotNull;
+import org.apache.hadoop.hbase.hbtop.field.Field;
+import org.apache.hadoop.hbase.hbtop.field.FieldValue;
+import org.apache.yetus.audience.InterfaceAudience;
+
+
+/**
+ * Represents a record of the metrics in the top screen.
+ */
+@InterfaceAudience.Private
+public class Record implements Map<Field, FieldValue> {
+
+  private Map<Field, FieldValue> values;
+
+  public static final class Entry implements Map.Entry<Field, FieldValue> {
+    private final Field key;
+    private FieldValue value;
+
+    private Entry(Field key, FieldValue value) {
+      this.key = Objects.requireNonNull(key);
+      this.value = Objects.requireNonNull(value);
+    }
+
+    @Override
+    public Field getKey() {
+      return key;
+    }
+
+    @Override
+    public FieldValue getValue() {
+      return value;
+    }
+
+    @Override
+    public FieldValue setValue(FieldValue value) {
+      FieldValue oldValue = this.value;
+      this.value = value;
+      return oldValue;
+    }
+  }
+
+  public static Entry entry(Field field, Object value) {
+    return new Entry(field, field.newValue(value));
+  }
+
+  public static Entry entry(Field field, FieldValue value) {
+    return new Entry(field, value);
+  }
+
+  public static Record ofEntries(List<Entry> entries) {
+    return ofEntries(entries.stream());
+  }
+
+  public static Record ofEntries(Entry... entries) {
+    return ofEntries(Stream.of(entries));
+  }
+
+  public static Record ofEntries(Stream<Entry> entries) {
+    return entries
+      .collect(Record::new, (r, e) -> r.put(e.getKey(), e.getValue()), (r1, r2) -> {});
+  }
+
+  public Record() {
+    this(new EnumMap<>(Field.class));
+  }
+
+  private Record(Map<Field, FieldValue> values) {
+    this.values = values;
+  }
+
+  @Override
+  public int size() {
+    return values.size();
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return values.isEmpty();
+  }
+
+  @Override
+  public boolean containsKey(Object key) {
+    return values.containsKey(key);
+  }
+
+  @Override
+  public boolean containsValue(Object value) {
+    return values.containsValue(value);
+  }
+
+  @Override
+  public FieldValue get(Object key) {
+    return values.get(key);
+  }
+
+  @Override
+  public FieldValue put(Field key, FieldValue value) {
+    return values.put(key, value);
+  }
+
+  public FieldValue put(Field key, Object value) {
+    return values.put(key, key.newValue(value));
+  }
+
+  @Override
+  public FieldValue remove(Object key) {
+    return values.remove(key);
+  }
+
+  @Override
+  public void putAll(@NotNull Map<? extends Field, ? extends FieldValue> m) {
+    values.putAll(m);
+  }
+
+  @Override
+  public void clear() {
+    values.clear();
```

Comment:
Ok if sloppy with values? Won't be multithreaded in console so should be fine

Reply:
Will make this object immutable, thanks.